### PR TITLE
feat: add manifest / multi-arch selection support

### PR DESCRIPTION
**Pull request by cdrage**
feat: add manifest / multi-arch selection support

### What does this PR do?

* Manifests are now listed, inspected and propagated within the Build
  page
* Simply select a manifest, select the arch and it will build

### Screenshot / video of UI

https://github.com/containers/podman-desktop-extension-bootc/assets/6422176/5a2c32a2-cddc-4180-9758-c7512d48bbf8

### What issues does this PR fix or reference?

Closes https://github.com/containers/podman-desktop-extension-bootc/issues/324

### How to test this PR?

Follow the video above, or do the following tests:

1. Build a manifest within Podman Desktop (select two architectures, and
   build a bootc image). Must use the latest Podman Desktop
2. Select the manifest within the extension and see it build.

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

### CODE REVIEW

**Comment by cdrage** [comment]
**IMPORTANT NOTES**

There are moving parts that need to be done before this merges:

- [ ] Added more tests (one is currently commented out)
- [ ] https://github.com/containers/podman-desktop-extension-bootc/issues/340 merged
- [ ] https://github.com/osbuild/images/pull/595 merged
- [ ] bootc-image-builder updated with latest osbuild/images changes
- [ ] update the image (it has been changed to a temporary one) in this PR to the updated bib image.

However, this is ready for review and testing (see changes in the last commit) 💯

File: packages/backend/src/container-utils.ts
Comment (deboer-tim) [inline packages/backend/src/container-utils.ts:336]: If no digests then doesn't this return an empty list? If so seems more straightforward to just use an if and return early.

File: packages/frontend/src/Build.spec.ts
Comment (deboer-tim) [inline packages/frontend/src/Build.spec.ts:83]: There is so much boilerplate/default values here I am wondering if it is a case to remove most of it and use `as unknown as ImageInspectInfo` :)

File: packages/backend/src/api-impl.ts
Comment (deboer-tim) [inline packages/backend/src/api-impl.ts:111]: The flow here is a little odd to me where this one case uses `continue`, but others set `includeImage` (or not). It would be more consistent to do similar logic in all cases, i.e. either use continue to avoid all invalid cases and always do a push at the end, or this case just doesn't set includeImage.

**Comment by cdrage** [comment]
Removed the WIP title and this is ready for a preliminary review.

However.. don't merge until https://github.com/osbuild/images/pull/595 is closed and completed, and this PR has an updated bib image.

**Comment by cdrage** [comment]
https://github.com/osbuild/bootc-image-builder/pull/397 has been merged!

This PR is ready for review!

Please throughly test building a manifest from within Podman Desktop, and test it out with bootc (build both arm64 and amd64 please!)

**Review by deboer-tim — APPROVED** [review]
Awesome work Charlie!

I was able to build several manifests using Podman Desktop 1.10, and they were as seamless (just twice as long!) as building for one platform.

I've verified the Build page automatically makes the correct platform(s) available as you switch between various images of either platform, or a multi-arch manifest.

I've verified that using a manifest and picking the ARM option builds and creates a usable image. Picking the AMD option clearly builds a different disk image and takes longer. I haven't had a chance to try running the built image on an AMD machine, but since that's not directly part of this PR I'm happy to approve already.

**Pull request merged**
